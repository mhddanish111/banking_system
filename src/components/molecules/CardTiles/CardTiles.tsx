import React from 'react';
import { split, get } from 'lodash';
import { Link } from 'react-router-dom';
import { cardNumberFormat } from '../../../utils';
import { Cards } from '../../../type/AccountType';
import './CardTiles.scss';

const CardTiles: React.FC<Cards> = ({ cards }: Cards) => {
  const { categoryName, accountNumber, expiryMonth, expiryYear, users } =
    cards || {};
  const getCategory = (): string => {
    const categoryData = split(categoryName, ' ', 1);
    return categoryData[0];
  };
  const firstName = get(users, ['0', 'firstName'], '');
  const lastName = get(users, ['0', 'lastName'], '');
  const month = expiryMonth < 10 ? `0${expiryMonth}` : expiryMonth;
  return (
    cards && (
      <div className="col-sm-3 col-md-4 col-lg-3 col-xs-12 card-tiles">
        <div className="col-xs-12 account-category-section">
          <div className="col-xs-12 end-xs">
            <span className="box">{getCategory()}</span>
          </div>
        </div>

        <div className="col-xs-12 middle-xs">
          <div className="box card-details">
            <span className="card-number">
              {accountNumber && cardNumberFormat(`${accountNumber}`)}
            </span>
          </div>
        </div>
        <div className="col-xs-12 card-exp">
          <div className="col-xs-6 start-xs">
            {(firstName || lastName) && (
              <span>{`${firstName} ${lastName}`}</span>
            )}
          </div>
          <div className="col-xs-6 end-xs">
            <span>{`Exp: ${month}/${expiryYear}`}</span>
          </div>
        </div>
        <div className="col-xs-12 card-bottom">
          <div className="col-xs-6 start-xs start-box">
            <Link to="/credit-card" className="box">
              Details
            </Link>
          </div>
          <div className="col-xs-6 end-xs">
            <Link to="/credit-card" className="box">
              Pay
            </Link>
          </div>
        </div>
      </div>
    )
  );
};

export default CardTiles;
