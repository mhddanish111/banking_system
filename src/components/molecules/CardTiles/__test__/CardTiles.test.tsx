import { shallow } from 'enzyme';
import CardTiles from '../CardTiles';

describe('<CardTiles />', () => {
  const cards = {
    type: 'Accounts',
    accountType: 'Savings Account',
    categoryName: 'Regular Account',
    accountNumber: 123456789,
    amount: 10000,
    debitCardCSVNumber: 123,
    debitCardNumber: 12312313123,
    expiryMonth: 1,
    expiryYear: 2023,
    limit: 0,
    maxAmount: 0,
    minAmount: 0,
    _id: '12345',
    charge: 0,
    users: [
      {
        _id: '12345',
        firstName: 'test',
        lastName: 'test',
      },
    ],
  };
  const wrapper = shallow(<CardTiles cards={cards} />);
  it('should render the CardTiles Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
