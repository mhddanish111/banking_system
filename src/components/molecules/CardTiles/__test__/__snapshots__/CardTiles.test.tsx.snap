// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardTiles /> should render the CardTiles Component correctly 1`] = `
<div
  className="col-sm-3 col-md-4 col-lg-3 col-xs-12 card-tiles"
>
  <div
    className="col-xs-12 account-category-section"
  >
    <div
      className="col-xs-12 end-xs"
    >
      <span
        className="box"
      >
        Regular
      </span>
    </div>
  </div>
  <div
    className="col-xs-12 middle-xs"
  >
    <div
      className="box card-details"
    >
      <span
        className="card-number"
      >
        1234 5678 9 
      </span>
    </div>
  </div>
  <div
    className="col-xs-12 card-exp"
  >
    <div
      className="col-xs-6 start-xs"
    >
      <span>
        test test
      </span>
    </div>
    <div
      className="col-xs-6 end-xs"
    >
      <span>
        Exp: 01/2023
      </span>
    </div>
  </div>
  <div
    className="col-xs-12 card-bottom"
  >
    <div
      className="col-xs-6 start-xs start-box"
    >
      <Link
        className="box"
        to="/credit-card"
      >
        Details
      </Link>
    </div>
    <div
      className="col-xs-6 end-xs"
    >
      <Link
        className="box"
        to="/credit-card"
      >
        Pay
      </Link>
    </div>
  </div>
</div>
`;
